### PR TITLE
Implement an example app

### PR DESCRIPTION
**Pull request by Bringoff**
@algodave fyi

A simple example app was implemented, as described in issue #3.

I tried to keep it simple in terms of UI, didn't use external dependencies and complicated approaches, because from my perspective the primary reason to have an example app is to showcase the library usage.

### CODE REVIEW

File: example/flutter_rest_data_example/lib/data/beer_model.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter_rest_data/flutter_rest_data.dart';
+import 'package:flutter_rest_data_example/data/ingredient_model.dart';
+
+class BeerModel extends JsonApiModel {
+  BeerModel(JsonApiDocument jsonApiDoc) : super(jsonApiDoc);
+
+  String get name => attributes!['name'];
+
```
Comment (Bringoff) [inline example/flutter_rest_data_example/lib/data/beer_model.dart:8]: @algodave 
I don't like this cast here, but `attributes` are nullable in the api at the moment. I'd rather make this property an empty map, if there is no attributes, but maybe that's just me.

Comment (algodave) [reply example/flutter_rest_data_example/lib/data/beer_model.dart:8]: Good catch! We definitely need to make `attributes` non nullable.

**Comment by algodave** [comment]
> @algodave added "Favourites" functionality, but this data is lost after reload. There is a bug in `addedByEndpoint` implementation that is addressed in #12

@Bringoff Those Beer APIs do not support any write operation, so it's totally ok to loose favorites after reload. Let's not even invoke `save()` on the beer record after clicking the star icon, because it obviously produces an exception in online mode.

**Comment by Bringoff** [comment]
> Let's not even invoke `save()` on the beer record after clicking the star icon

@algodave Fixed

**Pull request merged**